bin/NativeTests/BigIntTest.h

```diff
@@ -0,0 +1,42 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// This file contains stubs needed to make BigIntTest successfully compile and link as well
+// as a means to emulate behavior of objects that interact with BigInt class
+
+
+#include "..\..\lib\Common\Core\Api.cpp"
+#include "..\..\lib\Common\Common\NumberUtilities.cpp"
+
+namespace Js
+{
+    void Throw::FatalInternalError(long)
+    {
+        assert(false);
+    }
+
+    bool Throw::ReportAssert(char const *, unsigned int, char const *, char const *)
+    {
+        assert(false);
+        return true;
+    }
+
+    void Throw::LogAssert(void)
+    {
+        assert(false);
+    }
+
+}
+
+template <typename EncodedChar>
+double Js::NumberUtilities::StrToDbl(const EncodedChar *, const EncodedChar **, bool& )
+{
+    assert(false);
+    return 0.0;// don't care
+}
+
+
+#include "..\..\lib\Common\DataStructures\BigInt.h"
+#include "..\..\lib\Common\DataStructures\BigInt.cpp"
```

Comment:
I believe the preferred method for exposing internals like this to tests is using TestHooks -- it may make some of the code here cleaner, though it may add its own complexity of requiring an ad-hoc C interface to the data structure. @curtisman thoughts?

Reply:
Yes, TestHooks is a good approach too! I might refactor this class to make it more readable (in other PR) or create a new class. Haven't decided yet

---
In reply to: [214253986](https://github.com/Microsoft/ChakraCore/pull/5657#discussion_r214253986) [](ancestors = 214253986)